rs/tests/ckbtc/src/lib.rs

```diff
@@ -141,11 +106,62 @@ docker run  --name=bitcoind-node -d \
     let universal_vm = deployed_universal_vm.get_vm().unwrap();
     let btc_node_ipv6 = universal_vm.ipv6;
 
+    // Regtest bitcoin node listens on 18444
+    // docker bitcoind image uses 8332 for the rpc server
+    // https://en.bitcoinwiki.org/wiki/Running_Bitcoind
+    // nginx auto proxy setups SSL reverse proxy and forwards HTTP request on 443 to 9332.
+    println!("{}", deployed_universal_vm
+        .block_on_bash_script(&format!(
+            r#"
+# Create SSL cert using minica. IC-OS already supports the CA cert used here.
+mkdir /tmp/certs
+cd /tmp/certs
+cp /config/cert.pem minica.pem
+cp /config/key.pem minica-key.pem
+echo "Making certs directory in $(pwd) ..."
+docker load -i /config/minica.tar
+docker run -v "$(pwd)":/output minica:image -ip-addresses="{btc_node_ipv6}"
+sudo mv "{btc_node_ipv6}"/cert.pem localhost.crt
+sudo mv "{btc_node_ipv6}"/key.pem localhost.key
+sudo chmod 644 localhost*
+
+# Run nginx auto proxy
+docker run -d --name=proxy -e ENABLE_IPV6=true -e DEFAULT_HOST=localhost -p 80:80 -p {HTTPS_PORT}:443 \
+           -v /tmp/certs:/etc/nginx/certs -v /var/run/docker.sock:/tmp/docker.sock:ro \
+           nginxproxy/nginx-proxy
+
+# Setup bitcoin.conf and run bitcoind
+# The following variable assignment prevents the dollar sign in Rust's BITCOIND_RPC_AUTH string
+# from being interpreted by shell.
+BITCOIND_RPC_AUTH='{BITCOIND_RPC_AUTH}'
+cat >/tmp/bitcoin.conf <<END
+    regtest=1
+    debug=1
+    whitelist=::/0
+    fallbackfee=0.0002
+    rpcauth=$BITCOIND_RPC_AUTH
+END
+docker run  --name=bitcoind-node -d \
+  -e VIRTUAL_HOST=localhost -e VIRTUAL_PORT={BITCOIND_RPC_PORT} -v /tmp:/bitcoin/.bitcoin \
+  -p {BITCOIN_CLI_PORT}:{BITCOIN_CLI_PORT} -p {BITCOIND_RPC_PORT}:{BITCOIND_RPC_PORT} \
+  ghcr.io/dfinity/bitcoind@sha256:17c7dd21690f3be34630db7389d2f0bff14649e27a964afef03806a6d631e0f1
```

Comment:
Ideally this image is also moved in the UVM config image so that it doesn't have to be downloaded each time.

Reply:
Done.